psse/psse-model/src/main/java/com/powsybl/psse/model/io/LegacyTextReader.java

```diff
@@ -123,14 +127,38 @@ public String readRecordLine() throws IOException {
         return newLine.toString().trim();
     }
 
-    // all the lines beginning with "@!" are record lines defining the attribute fields
-    private static boolean isRecordLineDefiningTheAttributeFields(String line) {
-        return line.length() >= 2 && line.substring(0, 2).equals("@!");
+    private static String removeComment(String line) {
+        return line.replaceAll("('[^']*')|([^/']+)|(/.*)", "$1$2");
     }
 
-    private static String removeComment(String line) {
-        // Only outside quotes
-        return line.replaceAll("('[^']*')|(^/[^/]*)|(/[^/]*)", "$1$2");
+    // Compact spaces, remove spaces before the comma, and replace space with comma outside quoted text
+    private static String processText(String line) {
+        if (line == null || line.isEmpty()) {
+            return line;
+        }
+
+        StringBuilder result = new StringBuilder();
+        Pattern pattern = Pattern.compile("'[^']*'|[^']+");
+        Matcher matcher = pattern.matcher(line.trim());
+
+        while (matcher.find()) {
+            String part = matcher.group();
+            if (part.startsWith("'") && part.endsWith("'")) {
+                result.append(part);
+            } else {
+                // Outside quotes: process the txt
+                result.append(part.replaceAll("\\s+,", ",")
+                        .replaceAll(",\\s+", ",")
+                        .replaceAll("\\s+", ","));
+            }
+        }
+
+        return result.toString();
+    }
+
```

Comment:
maybe to concentrate all patterns related to the quote character in a single place

Define the pattern in the FileFormat class
`public static final Pattern LEGACY_QUOTED_OR_NON_QUOTED_TEXT = Pattern.compile("'[^']*'|[^']+");`
and use it here
`Pattern pattern = FileFormat.LEGACY_QUOTED_OR_NON_QUOTED_TEXT;`

We can also receive the quote character as a method argument and use it within the method

`private static String processText(String line, String quoteCharacter)` 

`if (part.startsWith(quoteCharacter) && part.endsWith(quoteCharacter))`

and the method must be used as

`Matcher m = FileFormat.LEGACY_TEXT_QUOTED_OR_WHITESPACE.matcher(processText(removeComment(line), String.valueOf(LEGACY_TEXT.getQuote())));`